h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = element.querySelector('.js-search-bar-input');
```

Comment:
We've changed the way that elements referenced by controllers are identified to avoid confusion with the identifiers that trigger upgrades.

The new method is to add a `data-ref="name"` attribute on the element and then use `this.refs.name` inside the controller. eg. `data-ref="searchBarInput"`, `this.refs.searchBarInput`